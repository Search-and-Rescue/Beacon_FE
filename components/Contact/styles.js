const styles = {
  contactContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  input: {
    backgroundColor: "#fff",
    borderColor: "black",
    borderWidth: 1,
    marginBottom: 10,
    padding: 10,
    width: "80%"
  },
  updateBtn: {
    borderColor: "black",
    borderWidth: 1,
    padding: 10,
    width: "80%"
  },
  updateBtnText: {
    textAlign: "center",
  }
}

export default styles;