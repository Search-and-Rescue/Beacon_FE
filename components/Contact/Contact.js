import React, { Component } from "react";
import { View, Text, TextInput, TouchableOpacity } from "react-native";
import styles from "./styles";

export default class Contact extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      phone: "",
      email: ""
    };
  }

  handleSubmit = () => {
    const { navigation } = this.props;
    navigation.navigate("ContactList");
    this.setState({
      make: "",
      model: "",
      year: "",
      color: "",
      license_plate: ""
    });
  };

  render() {
    return (
      <View style={styles.contactContainer}>
        <Text>Contact Page</Text>
        <TextInput
          placeholder="name"
          style={styles.input}
          onChangeText={text => this.setState({ name: text })}
          value={this.state.name}
          name="name"
        />
        <TextInput
          placeholder="phone"
          style={styles.input}
          onChangeText={text => this.setState({ phone: text })}
          value={this.state.phone}
          name="phone"
        />
        <TextInput
          placeholder="email"
          style={styles.input}
          onChangeText={text => this.setState({ email: text })}
          value={this.state.email}
          name="email"
        />
        <TouchableOpacity style={styles.updateBtn} onPress={this.handleSubmit}>
          <Text style={styles.updateBtnText}>Update</Text>
        </TouchableOpacity>
      </View>
    );
  }
}
