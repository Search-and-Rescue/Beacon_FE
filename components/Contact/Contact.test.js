import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import Contact from './Contact';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Contact', () => {

  beforeEach(() => {
    mockContact = {
      navigation: {
        state: {
          params: {
            contact: {
              id: 1,
              name: "Mom",
              phone: "123 456 7890",
              email: "mom@email.com"
            }
          }
        }
      }
    };
  });

  it('should match the snapshot with all of the data passed through', () => {
    expect(shallow(<Contact {...mockContact}/>)).toMatchSnapshot();
  });
});