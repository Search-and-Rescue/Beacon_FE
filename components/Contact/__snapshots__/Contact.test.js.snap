// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should match the snapshot with all of the data passed through 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginVertical": 40,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 12,
        "textAlign": "left",
        "width": "80%",
      }
    }
  >
    Name:
  </Text>
  <TextInput
    allowFontScaling={true}
    name="name"
    onChangeText={[Function]}
    placeholder="Jane D. Hiker"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "black",
        "borderWidth": 1,
        "marginBottom": 10,
        "padding": 10,
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value="Mom"
  />
  <Text
    style={
      Object {
        "fontSize": 12,
        "textAlign": "left",
        "width": "80%",
      }
    }
  >
    Phone number:
  </Text>
  <TextInput
    allowFontScaling={true}
    name="phone"
    onChangeText={[Function]}
    placeholder="(970)123-4567"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "black",
        "borderWidth": 1,
        "marginBottom": 10,
        "padding": 10,
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value="123 456 7890"
  />
  <Text
    style={
      Object {
        "fontSize": 12,
        "textAlign": "left",
        "width": "80%",
      }
    }
  >
    Email address:
  </Text>
  <TextInput
    allowFontScaling={true}
    name="email"
    onChangeText={[Function]}
    placeholder="jane@email.com"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "black",
        "borderWidth": 1,
        "marginBottom": 10,
        "padding": 10,
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value="mom@email.com"
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 1,
        "padding": 10,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      ADD
    </Text>
  </TouchableOpacity>
</View>
`;
